dhis2/maintenance-app
# feat(predictor): conditionally show category combo for predictors

This PR adds a form field for the outputCombo property on the predictor model. This field was implemented as a field-override component because there were some subtleties to take into account:
- Don't show all `categoryOptionCombos`, only the ones that apply to the current `dataElement`
- Only show the field if a `dataElement` actually has any `categoryOptionCombos`, other than default

I also decided to clean up the `src/config/field-overrides/predictor.js` file a bit. I've extracted some separate entities from that file and placed them in individual files in the folder `src/config/field-overrides/predictor/`. So please note that the changes to the following files don't need to be reviewed:
```
src/config/field-overrides/predictor/ExpressionDialog.js
src/config/field-overrides/predictor/ExpressionField.js
src/config/field-overrides/predictor/expressionStatusActions.js
src/config/field-overrides/predictor/expressionStatusStore.js
```

Or to put it differently, pretty much all the actual changes are in:
`src/config/field-overrides/predictor/DataElementCategoryOptionCombo.js`

To test:
1. Go to http://localhost:8081/#/edit/otherSection/predictor/add
1. Throttle your network speed to Slow 3G to behold the DropDownLoader in its full glory
1. Select "All other new" in the "Output data element" field
1. See the dropdown appear and load
1. Switch to another "Output data element" and see it reload, and hide if no categoryOptionCombos were found
1. Select a value and click save, then verify the value persisted
